src/radical/pilot/configs/resource_uva.json

```diff
@@ -0,0 +1,37 @@
+{
+    "rivanna":
+    {
+        "description"                 : "Heterogeneous community-model Linux cluster",
+        "notes"                       : "Access from registered UVA IP address. See https://www.rc.virginia.edu/userinfo/rivanna/login/",
+        "schemas"                     : ["local", "ssh"],
```

Comment:
The idea of `interactive` mode is that you run RCT app within an interactive job, thus allocation is done, and in this matter SAGA doesn't need to communicate with a batch script. But in this example RP ResourceManager is Slurm-oriented (uses a corresponding env variables to configure resource representation within RP)

```
"interactive"                 :
        {
            "job_manager_endpoint"    : "fork://localhost/",
            "filesystem_endpoint"     : "file://localhost/"
        },
```

Reply:
Update: small addition to @mtitov comment, you can use interactive mode in 2 ways:

1. As Mikhail mentioned as an interactive job example:
   `srun -n 1 -p xxx -t 01:00:00 etc..`  then you can run rp example from within that interactive job:
   `python rp_example.py`

or 

2.  You can create `sbatch` script and specify all of the resource required for your `rp_example` and execute the `sbatch script`

Reply:
`srun -n 1` would not be an effective demonstration. Also, The rivanna admins recommend submitting interactive jobs with the `ijob` wrapper.

I used `ijob -A kas_dev -p standard --time=1-00:00:00 --ntasks-per-node=8` and in `config.json`:
```
    "uva.rivanna" : {
        "project" : null,
        "queue"   : "standard",
        "schema"  : "interactive",
        "cores"   : 8,
        "gpus"    : 0
    }
```

It completes very quickly without indicating an error and without having done much.

Terminal output:
```
(py3_11) bash-4.2$ ./09_mpi_tasks.py uva.rivanna

================================================================================
 Getting Started (RP version 1.22.0)
================================================================================

new session: [rp.session.udc-aw29-24b.mei2n.019436.0002]                       \
database   : [mongodb://eirrgang:****@95.217.193.116/scalems]                 ok
read config                                                                   ok

--------------------------------------------------------------------------------
submit pilots

create pilot manager                                                          ok
submit 1 pilot(s)
        pilot.0000   uva.rivanna               8 cores       0 gpus           ok

--------------------------------------------------------------------------------
submit tasks

create task manager                                                           ok
create 2 task description(s)
        ..                                                                    ok
submit: ########################################################################

--------------------------------------------------------------------------------
gather results

wait  : exit requested

--------------------------------------------------------------------------------
finalize

closing session rp.session.udc-aw29-24b.mei2n.019436.0002              \
close task manager                                                            ok
close pilot manager                                                            \
wait for 1 pilot(s)
              0                                                               ok
                                                                              ok
session lifetime: 16.6s                                                       ok

--------------------------------------------------------------------------------

```

Files produced:
```
/scratch/mei2n/radical.pilot.sandbox/rp.session.udc-aw29-24b.mei2n.019436.0002/pilot.0000
/scratch/mei2n/radical.pilot.sandbox/rp.session.udc-aw29-24b.mei2n.019436.0002/pilot.0000/bootstrap_0.sh
/scratch/mei2n/radical.pilot.sandbox/rp.session.udc-aw29-24b.mei2n.019436.0002/pilot.0000/task.000000
/scratch/mei2n/radical.pilot.sandbox/rp.session.udc-aw29-24b.mei2n.019436.0002/pilot.0000/task.000000/09_mpi_tasks.sh
/scratch/mei2n/radical.pilot.sandbox/rp.session.udc-aw29-24b.mei2n.019436.0002/pilot.0000/radical-utils-env.sh
/scratch/mei2n/radical.pilot.sandbox/rp.session.udc-aw29-24b.mei2n.019436.0002/pilot.0000/agent.0.cfg
/scratch/mei2n/radical.pilot.sandbox/rp.session.udc-aw29-24b.mei2n.019436.0002/pilot.0000/task.000001
/scratch/mei2n/radical.pilot.sandbox/rp.session.udc-aw29-24b.mei2n.019436.0002/pilot.0000/task.000001/09_mpi_tasks.sh

Reply:
@eirrgang, thanks for checking this. `srun` was just a general example. Every machine is different, and I am not a user of Rivanna. 

Regarding RP, would it be possible to add the agent/client sandbox here, please? Thanks.

Reply:
with the local tests using FORK (`"job_manager_endpoint"    : "fork://localhost/"`) - SAGA doesn't support attribute `queue`, with setting it to `null` should work
```
BadParameter: 'JobDescription.Queue' (standard) not supported by radical.saga.adaptors.shell_job
```

Reply:
To clarify, are you suggesting an edit like the following?
```
"interactive"                 :
        {
            "job_manager_endpoint"    : null,
            "filesystem_endpoint"     : "file://localhost/"
        },
```
Or are you saying that the `config.json` needs to be edited? i.e.
```
    "uva.rivanna" : {
        "project" : null,
        "queue"   : null,
        "schema"  : "interactive",
        "cores"   : 8,
        "gpus"    : 0
    }

Reply:
I had to set `default_queue` to `null` in the resource definition, as well, to avoid the same error. If `default_queue` cannot be set per scheme, then either `default_queue` needs to be removed from the resource definition or we cannot add `"interactive"`.

However, the session hung in when the bootstrap encountered errors trying to rebuild the virtual env. That seems like a dangerous bug, since there was no indication of problems to the caller. (Inspecting the sandbox, one sees `ERROR: rp installation incomplete?` in the `bootstrap_0.out`)

I removed the venv but neglected to remove the `ve.uva.rivanna.1.22.0.lock` file, so a resubmission again hung without feedback to the caller. I wonder if the necessity of the lock could be removed, but shouldn't RP just assume an error has occurred if the lock exists during prepare_venv?

Removing the lock file allowed the `09_mpi_tasks` job to proceed, eventually producing
```
  * task.000000: DONE, exit:   0, ranks: udc-aw29-23a 0:0/2 @ 0/40 : 0/1
udc-aw29-23a 0:1/2 @ 0/40 : 0/1
udc-aw29-23a 0:0/2 @ 0/40 : 0/1
udc-aw29-23a 0:1/2 @ 0/40 : 0/1
udc-aw29-23a 0:1/2 @ 0/40 : 0/1
udc-aw29-23a 0:0/2 @ 0/40 : 0/1

caught Exception: missing rank 1:0/2 (['0:0/2', '0:1/2', '0:0/2', '0:1/2', '0:1/2', '0:0/2'])
```

This looks like it might be a logic error in the example script itself. If someone wants to take a look, the directory on rivanna is `/scratch/mei2n/radical.pilot.sandbox/rp.session.udc-aw29-23a.mei2n.019437.0003`. However, per above, I don't see that the "interactive" scheme is useful if it is incompatible with the other schemes by way of the `default_queue` parameter, and I can't troubleshoot this any more right now.

For the record: It is very frustrating that errors like this do not have any Pythonic feedback to the caller and do not even appear in the `.err` files.

Reply:
> I had to set default_queue to null in the resource definition, as well, to avoid the same error. If default_queue cannot be set per scheme, then either default_queue needs to be removed from the resource definition or we cannot add "interactive".

this is resolved in SAGA, thus keep `default_queue` within the configuration, which will be used for non-interactive modes and will not break interactive one.

Reply:
> this is resolved in SAGA

To be precise, SAGA v1.22 was just released which will not raise on unsupported attributes anymore.